### ISSUE TITLE
Migrate from legacy life cycles - phase 2

### ISSUE THREAD

**Issue by OscarWang114** [wasedatime/wasedatime-web#38]
This is a future step after #30 is completed. Low priority for now.

- Read https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#migrating-from-legacy-lifecycles
- Replace UNSAFE_ComponentWillMount,  UNSAFE_ComponentWillReceiveProps,  UNSAFE_ComponentWillUpdate with new life cycles: getDerivedStateFromProps and getSnapshotBeforeUpdate. Be careful! Simply changing the name is not enough.
